route/build/src/pio_utils.f90

```diff
@@ -503,6 +504,41 @@ END SUBROUTINE def_var
   ! Writing routine
   ! -----------------------------
 
+  ! ---------------------------------------------------------------
+  ! write attributes
+  subroutine put_attr(pioFileDesc,   &  ! input: netcdf file handle id
+                      varid,         &  ! input: netcdf var id
+                      attname,       &  ! input: attribute name
+                      attval,        &  ! input: attribute values
+                      ierr, message)    ! output: error control
+    implicit none
+    ! Argument variables:
+    type(file_desc_t),  intent(inout) :: pioFileDesc  ! pio file handle
+    integer(i4b)     ,  intent(in)    :: varid        ! netcdf var id
+    character(len=*) ,  intent(in)    :: attname      ! netcdf attrib
+    class(*)         ,  intent(in)    :: attval       ! netcdf attrib value
+    integer(i4b)     ,  intent(out)   :: ierr         ! error code
+    character(*)     ,  intent(out)   :: message      ! error message
+
+    ierr=0; message='put_attr/'
+
+    select type (attval)
+      type is (integer(i4b))
```

Comment:
Probably should make these the default, so it would be done the same for anything other than character.

Reply:
Hi @ekluzek ,   Looks like the compilers do not like if these are changed to

```
 525     select type (attval)
 526       type is (character(len=*))
 527         ierr = PIO_put_att(pioFileDesc, varid, trim(attname), trim(attval))
 528       class default  ! real(dp), real(sp), integer(i4b), integer(i8b)
 529         ierr = PIO_put_att(pioFileDesc, varid, trim(attname), attval)
 530     end select
```

`error #8769: If the actual argument is unlimited polymorphic, the corresponding dummy argument must also be unlimited polymorphic.  `
 this error points to line 529.    Looking for some more information, but not really finding anything.  

I will keep this as is, and if I can find any way to change (simplify) these lines, I will change.

Reply:
OK, it's really not that bad as it is.

The error message says that the PIO interface would have to change. So that's beyond our scope we need here.